### PR TITLE
Mobile Select - "Fullscreen" mode

### PR DESCRIPTION
**Pull request by TomTirapani**
Allow showing mobile selects in "fullscreen" mode to facilitate space for the device keyboard.

See issue: https://github.com/xh/hoist-react/issues/2274
See toolbox PR: https://github.com/xh/toolbox/pull/473

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: mobile/cmp/input/Select.js [hunk 1]
```diff
@@ -105,6 +116,9 @@ Select.propTypes = {
     /** Text to display when control is empty. */
     placeholder: PT.string,
 
+    /** Text to display in header when in fullscreen mode. */
```
Comment (lbwexler) [inline mobile/cmp/input/Select.js:119]: nit:  might just call this 'title' and mention that it is currently used in `fullScreen` mode

File: mobile/cmp/input/Select.js [hunk 2]
```diff
@@ -385,14 +433,51 @@ const cmp = hoistCmp.factory(
...
+                    className: 'xh-select__fullscreen-toolbar',
+                    items: [
+                        button({
+                            icon: Icon.chevronLeft(),
+                            onClick: () => model.setFullscreen(false)
+                        }),
+                        span(title)
+                    ]
+                }),
+                children
```
Comment (lbwexler) [inline mobile/cmp/input/Select.js:479]: realize there is only a single child, but this is confusing.  Might either spread with ...children, or use React.Children.only(children)

**Review by lbwexler — APPROVED** [review]
just some super-minor nits.  Looks great

**Comment by TomTirapani** [comment]
Thanks for taking a look Lee - I've updated based on your suggestions.

What did you think about the concern raised on the issue re: inability to automatically bring up keyboard on device?
https://github.com/xh/hoist-react/issues/2274

File: mobile/cmp/input/Select.js [hunk 3]
```diff
@@ -42,18 +47,25 @@ Select.propTypes = {
...
-     * showing the dropdown menu on click. Should be used with caution, as mobile keyboard can
-     * cause undesirable results.
+     * showing the dropdown menu on click. Usually used with enableFullscreen, to ensure access
+     * to the mobile device keyboard.
      */
     enableFilter: PT.bool,
 
+    /**
+     * True to render the select control in a full-screen modal dialog when focused.
+     * Usually applicable when enableCreate|enableFilter, as we can guarantee the control
```
Comment (amcclain) [inline mobile/cmp/input/Select.js:64]: Suggest replacing "Usually..." in these comments with "Recommended for use with..."

File: mobile/cmp/input/Select.scss [hunk 4]
```diff
@@ -81,3 +81,37 @@
...
+
+  .xh-select {
+    margin: var(--xh-pad-px);
+
+    &__fullscreen-wrapper {
+      position: absolute !important;
+      top: 0;
+      left: 0;
+      right: 0;
+      bottom: 0;
```
Comment (amcclain) [inline mobile/cmp/input/Select.scss:99]: Is it standard to specify right and bottom like this (in addition to top/left) for an absolutely positioned element?

File: mobile/cmp/input/Select.scss [hunk 5]
```diff
@@ -81,3 +81,37 @@
...
+      background: var(--xh-bg);
+    }
+
+    &__fullscreen-toolbar {
+      background-color: var(--xh-appbar-bg);
+      color: var(--xh-appbar-title-color);
+      padding: var(--xh-pad-half-px) var(--xh-pad-px) var(--xh-pad-half-px) 0;
```
Comment (amcclain) [inline mobile/cmp/input/Select.scss:106]: Seeing a funny artifact when testing on my iPhone 12, where the back button is stretched against the left edge of the toolbar.  Weird, as I see we have left padding specified...

IMG_0615

**Review by amcclain — COMMENTED** [review]
I tested on my iPhone and the keyboard is showing up automatically OK for me - is that what you guys were seeing? (This is in response to https://github.com/xh/hoist-react/issues/2274#issuecomment-810458552).

I had been thinking of the totally dedicated component with the text box and the grid, etc. but this seems to work well and has several very big benefits in terms of re-using `select` and all of its various API options, instead of re-inventing a more minimal component. 

I also like that the change is restricted to the mobile select component only, which gives me more confidence in merging, and I think it remains a usability win for working with longer select lists on the phone.

Let me know if the keyboard issue is still current / requires more investigation, or if there are any other causes for hesitation. Otherwise I'd be in favor of adding a changelog entry and merging.

File: mobile/cmp/input/Select.scss (hunk 4 above)
Comment (TomTirapani) [reply mobile/cmp/input/Select.scss:99]: It's another way of making an absolute positioned element fill its container. We could also give it `top: 0; left: 0; width: 100%; height: 100%;`, but I think that might have some unexpected behaviour in some circumstances (i.e. container is floating).

https://rajatamil.medium.com/css-make-a-div-full-screen-402f9a935d15

We're using it in many other places throughout the code (CodeInput, TreeMap, Mask, etc)

File: mobile/cmp/input/Select.scss (hunk 5 above)
Comment (TomTirapani) [reply mobile/cmp/input/Select.scss:106]: Actually, that button is supposed to be transparent (similar to the main app bar colors)... will update the styling. Thanks for catching!

**Comment by TomTirapani** [comment]
Thanks for taking a look Anselm!

I'm happy to merge at this point - agree there are some benefits to building on top of `Select` rather than a new component. I've pushed some tweaks based on your feedback and added a CHANGELOG entry.

File: mobile/cmp/input/Select.scss (hunk 4 above)
Comment (amcclain) [reply mobile/cmp/input/Select.scss:99]: Nice, thanks for the info - will (attempt to) remember it for when I need to do the same!

**Comment by amcclain** [comment]
Great - glad this got merged in. 👍

**Pull request merged**